core/Settings/Setting.php

```diff
@@ -226,7 +226,7 @@ public function setValue($value)
             $value = call_user_func($config->transform, $value, $this);
         }
 
-        if (isset($this->type)) {
+        if (isset($this->type) && !is_null($value)) {
```

Comment:
This is needed as otherwise `null` values would be casted to `0` or empty string